cmd/juju/cloud/add_test.go

```diff
@@ -499,7 +501,47 @@ func (*addSuite) TestInteractiveAddCloud_PromptForNameIsCorrect(c *gc.C) {
 	// Running the command will return an error because we only give
 	// enough input to get to the prompt we care about checking. This
 	// test ignores this error.
-	command.Run(ctx)
+	err := command.Run(ctx)
+	c.Assert(errors.Cause(err), gc.Equals, io.EOF)
 
 	c.Check(out.String(), gc.Matches, "(?s).+Enter a name for your manual cloud: .*")
 }
+
+func (*addSuite) TestSpecifyingCloudFileThroughFlag_CorrectlySetsMemberVar(c *gc.C) {
+	command := cloud.NewAddCloudCommand(nil)
+	runCmd := func() {
+		testing.RunCommand(c, command, "garage-maas", "-f", "fake.yaml")
+	}
+	c.Assert(runCmd, gc.PanicMatches, "runtime error: invalid memory address or nil pointer dereference")
+	c.Check(command.CloudFile, gc.Equals, "fake.yaml")
+}
+
+func (*addSuite) TestSpecifyingCloudFileThroughFlagAndArgument_Errors(c *gc.C) {
+	command := cloud.NewAddCloudCommand(nil)
+	_, err := testing.RunCommand(c, command, "garage-maas", "-f", "fake.yaml", "foo.yaml")
+	c.Check(err, gc.ErrorMatches, "cannot specify cloud file with flag and argument")
+}
+
+func (*addSuite) TestSpecifyCloudName_ProvidedNamedUtilizedDuringInteractive(c *gc.C) {
+	fake := newFakeCloudMetadataStore()
+	fake.Call("PublicCloudMetadata", []string(nil)).Returns(map[string]cloudfile.Cloud{}, false, nil)
+	fake.Call("PersonalCloudMetadata").Returns(homestackMetadata(), nil)
+	command := cloud.NewAddCloudCommand(fake)
+	command.Cloud = "foo-name"
+
+	var out bytes.Buffer
+	ctx := &cmd.Context{
+		Stdout: &out,
+		Stderr: ioutil.Discard,
+		Stdin: strings.NewReader("" +
```

Comment:
can you fix the input so that we successfully run the command entirely?  That way if we move the name query later in the sequence, we're still ensured that this test will fail if the name gets queried later.